pkg/api/v1/resource/helpers.go

```diff
@@ -39,16 +42,23 @@ func GetResourceRequestQuantity(pod *v1.Pod, resourceName v1.ResourceName) resou
 		requestQuantity = resource.Quantity{Format: resource.DecimalSI}
 	}
 
-	for _, container := range pod.Spec.Containers {
-		if rQuantity, ok := container.Resources.Requests[resourceName]; ok {
+	// Supported pod level resources will be used instead of container level ones when available
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) && resourcehelper.IsPodLevelResourcesSet(pod) && resourcehelper.IsSupportedPodLevelResource(resourceName) {
+		if rQuantity, ok := pod.Spec.Resources.Requests[resourceName]; ok {
 			requestQuantity.Add(rQuantity)
 		}
-	}
+	} else {
+		for _, container := range pod.Spec.Containers {
+			if rQuantity, ok := container.Resources.Requests[resourceName]; ok {
+				requestQuantity.Add(rQuantity)
+			}
+		}
 
-	for _, container := range pod.Spec.InitContainers {
-		if rQuantity, ok := container.Resources.Requests[resourceName]; ok {
-			if requestQuantity.Cmp(rQuantity) < 0 {
-				requestQuantity = rQuantity.DeepCopy()
+		for _, container := range pod.Spec.InitContainers {
```

Comment:
This isn't factoring in sidecars. This function needs to replicate the logic from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-helpers/resource/helpers.go#L122.

It might be better to avoid duplicating the logic, and add a resource filter to PodResourcesOptions.

Reply:
What do you think about just calling `AggregateContainerRequests`?
```
requestQuantity.Add(resourcehelper.AggregateContainerRequests(pod, resourcehelper.PodResourcesOptions{})[resourceName])
```

Reply:
Done, thank you. I removed duplicated logic and added an option `opts.SkipContainerLevelResources` to `func PodRequests`, so container aggregation could be skipped when called from `func GetResourceRequestQuantity`.